packages/SwingSet/test/test-promises.js

```diff
@@ -251,24 +249,36 @@ test('local promises are rejected by vat upgrade', async t => {
     const err = c.kpResolution(kpid);
     throw kunser(err);
   };
-  const messageVat = (name, methodName, args) =>
-    run('messageVat', [{ name, methodName, args }]);
-  // eslint-disable-next-line no-unused-vars
-  const messageObject = (presence, methodName, args) =>
-    run('messageVatObject', [{ presence, methodName, args }]);
+
+  const messageToVat = async (vatName, method, ...args) => {
+    assert(Array.isArray(args));
+    const kpid = c.queueToVatRoot(vatName, method, args);
+    return awaitRun(kpid);
+  };
+  const messageToObject = async (presence, method, ...args) => {
+    assert(Array.isArray(args));
+    const kpid = c.queueToVatObject(presence, method, args);
+    return awaitRun(kpid);
+  };
 
   const S = Symbol.for('passable');
-  await run('createVat', [{ name: 'watcher', bundleCapName: 'watcher' }]);
-  await messageVat('watcher', 'watchLocalPromise', ['orphaned']);
-  await messageVat('watcher', 'watchLocalPromise', ['fulfilled', S]);
-  await messageVat('watcher', 'watchLocalPromise', ['rejected', undefined, S]);
-  const v1Settlements = await messageVat('watcher', 'getSettlements');
+  const watcher = await messageToVat('bootstrap', 'createVat', {
+    name: 'watcher',
+    bundleCapName: 'watcher',
+  });
+  await messageToObject(watcher, 'watchLocalPromise', 'orphaned');
+  await messageToObject(watcher, 'watchLocalPromise', 'fulfilled', S);
+  await messageToObject(watcher, 'watchLocalPromise', 'rejected', undefined, S);
+  const v1Settlements = await messageToObject(watcher, 'getSettlements');
```

Comment:
Something `E`-like that was syntactically friendlier might be a good idea; it would certainly make the code easier to write and easier on the eyes, though probably out of scope for this PR.  The later PR we have planned for cleaning up the support code and putting it into a proper library of its own might be the place for that.  I'm a bit nervous about `EV` because it leads you down the garden path into thinking you can use it just like you use `E`, which is untrue in important ways.  (Also, I should probably put a big giant warning comment on the definition of `EV` and its associated functions, explaining the important hazard of using them, which possibly even their author(s) might be unaware of.)

Reply:
Actually, it strikes me that `messageToObject` and its variants have the same hazard as `EV`, which might well motivate putting all that stuff in one place if only to have a common place to hang the warnings.